### DISCUSSION TITLE
Google creds stop working after a few months

### DISCUSSION THREAD

**Discussion by sjd23** [emailjs-com/help#120] [Q&A]
After some unknown time period (months? a year?), with no changes or modifications to the connected apps on my Gmail, I start to see this error in the logs:  "
Gmail_API: Invalid grant. Please reconnect your Gmail account" 
![image](https://github.com/emailjs-com/help/assets/145370727/5ac595f0-cf7c-4e33-8f23-fab85237afff)

1. All I do to resolve this is refresh the connection. Since there are no cred changes or anything else, this seems like it will always happen no matter what we do. Therefore, shouldn't there be some sort of scheduled job that does this automatically or at least some static set of shared information on how often we should manually refresh/reset our connection to Gmail?

2. How can I get an email or some sort of notification that this error is happening? We have lost very important contact information because of this, and I only ever know because I randomly test my site and see that it isn't working. Then I go into the event logs and see that we've missed potentially months of important emails.

**Comment by xr0master** [comment]
1. Gmail can revoke access at any time, usually due to changes in Gmail settings or suspicious behavior. But this happens very rarely. Unfortunately, this process cannot be automated as it requires real user action.
2. The EmailJS service sends a notification if there are any issues with the connected provider. Please make sure that your email address for notification is configured correctly. You can also check your SPAM folder, just in case.

**Reply by sjd23** [reply]
Thanks for the quick response. I definitely understand point # 1. As for # 2, I receive all other emails from EmailJS successfully so I don't think that spam is the issue. Is there a notification setting for this somewhere that I can check? I don't see too many settings in account/profile so I figured it'd use the same email I see here:

![image](https://github.com/emailjs-com/help/assets/145370727/e692182c-15e0-474e-bc1e-f8511b84be0d)

**Reply by xr0master** [reply]
right. the place is correct

**Reply by sjd23** [reply]
Ok then in that case I can confirm we do not receive emails for this currently. I hit the test button a few times when this happened last time and checked my spam, inbox, etc. There was nothing there although the error was logged.
